### PR TITLE
Added docstrings to base_device.py

### PR DESCRIPTION
**Pull request by qduk**

### CODE REVIEW

File: pyntc/devices/base_device.py [hunk 1]
```diff
@@ -9,6 +9,11 @@
 
 
 def fix_docs(cls):
+    """Create docstring at runtime.
+
+    Returns:
+        class: Returns the class passed in.
+    """
```
Comment (qduk) [inline pyntc/devices/base_device.py:16]: I discussed the appropriate way to document returning the class that was passed with Dr. X. We weren't too sure the proper verbiage.  As this function will be going away after all docsrings are implemented I just put a very basic docstring.

Comment (mundruid) [reply pyntc/devices/base_device.py:16]: I still have not found anything relevant.

**Review by mundruid — APPROVED** [review]

File: pyntc/devices/base_device.py [hunk 2]
```diff
@@ -156,7 +170,7 @@ def hostname(self):
 
     @property
     @abc.abstractmethod
-    def fqdn(self):
+    def fqdn(self):  # noqa:  D403
```
Comment (jmcgill298) [inline pyntc/devices/base_device.py:173]:
```suggestion
    def fqdn(self):
```

File: pyntc/devices/base_device.py [hunk 3]
```diff
@@ -156,7 +170,7 @@ def hostname(self):
...
         """fqdn name string property, part of device facts.
```
Comment (jmcgill298) [inline pyntc/devices/base_device.py:174]:
```suggestion
        """
        Get FQDN of the device.
```

File: pyntc/devices/base_device.py [hunk 4]
```diff
@@ -176,7 +190,7 @@ def uptime_string(self):
 
     @property
     @abc.abstractmethod
-    def serial_number(self):
+    def serial_number(self):  # noqa:  D401
         """Serial number string property, part of device facts.
```
Comment (jmcgill298) [inline pyntc/devices/base_device.py:194]:
```suggestion
        """
        Get serial number of the device.
```

File: pyntc/devices/base_device.py [hunk 5]
```diff
@@ -205,6 +219,7 @@ def vlans(self):
         raise NotImplementedError
 
     def facts(self):
+        """Gather facts about device."""
```
Comment (jmcgill298) [inline pyntc/devices/base_device.py:222]:
```suggestion
        """
        DEPRECATED - Use individual properties to get facts."
        ""
```

File: pyntc/devices/base_device.py [hunk 6]
```diff
@@ -205,6 +219,7 @@ def vlans(self):
         raise NotImplementedError
 
     def facts(self):
```
Comment (jmcgill298) [inline pyntc/devices/base_device.py:221]:
```suggestion
    def facts(self):  # noqa: D403
```

**Review by jmcgill298 — CHANGES_REQUESTED** [review]

File: pyntc/devices/base_device.py [hunk 7]
```diff
@@ -176,7 +190,7 @@ def uptime_string(self):
 
     @property
     @abc.abstractmethod
-    def serial_number(self):
+    def serial_number(self):  # noqa:  D401
```
Comment (jmcgill298) [inline pyntc/devices/base_device.py:193]:
```suggestion
    def serial_number(self):
```

**Review by jmcgill298 — APPROVED** [review]

File: pyntc/devices/base_device.py [hunk 8]
```diff
@@ -204,7 +220,10 @@ def vlans(self):
         """
         raise NotImplementedError
 
-    def facts(self):
+    def facts(self):  # noqa: D403
+        """
+        DEPRECATED - Use individual properties to get facts."
+        ""
```
Comment (jmcgill298) [inline pyntc/devices/base_device.py:226]:
```suggestion
        """
```

Comment (jmcgill298) [inline pyntc/devices/base_device.py:226]:
```suggestion
        DEPRECATED - Use individual properties to get facts.
        """
```

**Pull request merged**